spring-integration-file/src/main/java/org/springframework/integration/file/NoopNewFileCallback.java

```diff
@@ -0,0 +1,33 @@
+/*
+ * Copyright 2002-2018 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.integration.file;
+
+import java.io.File;
+
+/**
+ * A {@link NewFileCallback} which does nothing for new files.
+ *
+ * @author Alen Turkovic
+ */
+public class NoopNewFileCallback implements NewFileCallback {
```

Comment:
I don't see reason in this. The simple `if (this.newFileCallback != null)` is fully enough.

Reply:
Will remove.